src/util/parameter_transfer.rs

```diff
@@ -180,7 +182,7 @@ mod tests {
 
         // Verify final values
         for p in 0..PARAMETERS {
-            assert!((0..THREADS).any(|t| results[t][p] == values[p]));
+            assert!((0..THREADS).any(|t| (results[t][p] - values[p]).abs() < std::f32::EPSILON));
```

Comment:
The test must use actual equality here, not the approximate one. It tests that the exact value present was set by at least one thread.

Reply:
Done (along with other fuzzy float compares)